util-test/src/main/scala/com/twitter/util/testing/ArgumentCapture.scala

```diff
@@ -0,0 +1,115 @@
+package com.twitter.util.testing
+
+import java.util.{List => JList}
+import org.mockito.ArgumentCaptor
+import org.mockito.exceptions.Reporter
+import scala.collection.JavaConversions._
+import scala.reflect._
+
+trait ArgumentCapture {
+  /**
+   * Enables capturingOne to be implemented over capturingAll with the same behavior as ArgumentCaptor.getValue
+   */
+  private[this] def noArgWasCaptured: Nothing = {
+    new Reporter().noArgumentValueWasCaptured() // this always throws an exception
+    throw new RuntimeException("this should be unreachable, but allows the method to be of type Nothing")
+  }
+
+  private[this] def zip4[T, U, V, W](args: (JList[T], JList[U], JList[V], JList[W])): Seq[(T, U, V, W)] = {
```

Comment:
Yeah, I heard that pants is weird about macros?  Not sure whether that's apocryphal or not.  This is probably fine for now.